### PR TITLE
Implement saving result components to database

### LINKED ISSUE #654: Implement Result.persist method
This issue is dedicated to implementing the `Result.persist` method. There are several auxiliary methods that need to be implemented in order to accomplish this. Planning documentation can be found [here](https://docs.google.com/document/d/1uaqhBQ6PffsxWHeUjFx2JWd6xGG2-j4lLJbMeJmJG3A/edit#).

**Acceptance criteria**

Implement the following methods / functions:
- [ ] `Result.persist` which is analogous to `Result.save`.
    - [x] `insert_lattices_data`
    - [x] `insert_electrons_data`
    - [x] `insert_electron_dependency_data`
    - [x] `update_lattices_data`
    - [x] `update_electrons_data`
    - [ ] `are_electron_dependencies_added`
    - [x] `get_electron_type`
    - [ ] connecting lattice and sublattices via electron_id. 
- [ ] Keep design doc updated if changes are required.
- [ ] Add unit tests with coverage >= 90%.

### PR DESCRIPTION
**Pull request by FyzHsn**
Fixes #654 

- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by FyzHsn** [comment]
@venkatBala Can you give review and merge this? But keep the issue open, I need to implement Result.persist before this issue can be fully closed.

File: covalent/_data_store/models.py [hunk 1]
```diff
@@ -22,35 +22,15 @@
 Models for the workflows db. Based on schema v9
 """
 
-import enum
-
-from sqlalchemy import BigInteger, Column, DateTime, Enum, ForeignKey, Integer, String, Text
+from sqlalchemy import BigInteger, Column, DateTime, ForeignKey, Integer, String, Text
```
Comment (venkatBala) [inline covalent/_data_store/models.py:25]: Is `BigInteger` still beused? if not we can consider not importing it?

**Comment by venkatBala** [comment]
do empty files also need the copyright header?

**Comment by FyzHsn** [comment]
> do empty files also need the copyright header?

I figured it can't hurt. But you're right it's not necessary. Would you like me to remove it?

**Comment by venkatBala** [comment]
> > do empty files also need the copyright header?
> 
> I figured it can't hurt. But you're right it's not necessary. Would you like me to remove it?

I am not sure whether is necessary or not either... was just wondering if the copyright info needs to be put in tests/covalent_tests/results_manager_tests/__init__.py... any best practices?

**Review by venkatBala — APPROVED** [review]

**Pull request merged**